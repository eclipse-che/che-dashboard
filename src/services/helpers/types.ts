/*
 * Copyright (c) 2018-2020 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { AlertVariant } from '@patternfly/react-core';
import * as React from 'react';

export interface AlertItem {
  key: string;
  title: string;
  variant: AlertVariant;
  children?: React.ReactNode;
}

export interface FactoryResolver {
  v: string;
  source: string;
  devfile: api.che.workspace.devfile.Devfile
}

export enum WorkspaceStatus {
  RUNNING = 1,
  STOPPING,
  STOPPED,
  STARTING,
  PAUSED,
  ERROR,
}

export enum IdeLoaderTabs {
  Progress = 0,
  Logs = 1,
}
