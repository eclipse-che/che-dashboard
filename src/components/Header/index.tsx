/*
 * Copyright (c) 2018-2020 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { PageSection, PageSectionVariants, Text, TextContent, TextVariants } from '@patternfly/react-core';
import WorkspaceStatusLabel from '../WorkspaceStatusLabel';

const SECTION_THEME = PageSectionVariants.light;

type Props = {
  status?: string;
  title: string;
};

class Header extends React.PureComponent<Props> {

  public render(): React.ReactElement {
    const { title, status } = this.props;

    return (
      <PageSection variant={SECTION_THEME}>
        <TextContent>
          <Text component={TextVariants.h1}>
            {title}{status && (
              <WorkspaceStatusLabel status={status} />
            )}
          </Text>
        </TextContent>
      </PageSection>
    );
  }
}

export default Header;
