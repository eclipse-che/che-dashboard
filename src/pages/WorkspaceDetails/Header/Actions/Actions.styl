.workspace-action-selector
  button
    border-radius 3px
  ul
    padding 0
