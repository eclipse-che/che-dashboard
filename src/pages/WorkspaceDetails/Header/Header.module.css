.workspaceDetailsHeader {
  padding-bottom: 0;
}

.workspaceDetailsHeader li:first-child {
  margin-right: 0;
}

.workspaceDetailsHeader li:first-child a {
  margin-right: 8px;
}

.workspaceDetailsHeader h1 {
  padding-top: 15px;
  line-height: 25px;
}
