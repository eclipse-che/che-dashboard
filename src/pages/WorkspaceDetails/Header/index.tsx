/*
 * Copyright (c) 2018-2020 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { PageSection, Text, TextVariants, FlexItem, Flex, PageSectionVariants } from '@patternfly/react-core';
import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';
import { ROUTE } from '../../../route.enum';
import WorkspaceStatusLabel from '../../../components/WorkspaceStatusLabel';

import styles from './Header.module.css';

type Props = {
  status: string | undefined;
  workspaceName: string;
};

class Header extends React.PureComponent<Props> {

  public render(): React.ReactElement {
    const { workspaceName, status, children } = this.props;

    return (
      <PageSection variant={PageSectionVariants.light} className={styles.workspaceDetailsHeader}>
        <Breadcrumb>
          <BreadcrumbItem to={`./#${ROUTE.WORKSPACES}`}>
            Workspaces
          </BreadcrumbItem>
          <BreadcrumbItem isActive>{workspaceName}</BreadcrumbItem>
        </Breadcrumb>
        <Flex>
          <FlexItem>
            <Text component={TextVariants.h1}>
              {workspaceName}
              <WorkspaceStatusLabel status={status} />
            </Text>
          </FlexItem>
          <FlexItem align={{ default: 'alignRight' }}>
            {children}
          </FlexItem>
        </Flex>
      </PageSection>
    );
  }
}

export default Header;
