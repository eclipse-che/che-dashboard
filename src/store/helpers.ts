/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FactoryResolver } from '../services/helpers/types';
import { isDevfileV2 } from '../services/workspaceAdapter';

const PROJECT_NAME_MAX_LENGTH = 63;

type ProjectV2Source = {
  name: string;
  git: {
    remotes: { origin: string };
    checkoutFrom?: { revision: string };
  }
};

/**
 * Creates a new state object.
 * @param state a store state, e.g workspaces, plugins.
 * @param partial a slice of a store state
 */
export function createState<T>(state: T, partial: Partial<T>): T {
  return Object.assign({}, state, partial);
}

/**
 * Returns a devfile from the FactoryResolver object.
 * @param data a FactoryResolver object.
 */
export function getDevfile(data: FactoryResolver): api.che.workspace.devfile.Devfile {
  let devfile = data.devfile;

  if (isDevfileV2(devfile)) {
    devfile = Object.assign(devfile, { components });
    // add a default project
    const projects: ProjectV2Source[] = [];
    const scmInfo = data['scm_info'];
    if (!devfile.projects?.length && scmInfo) {
      const origin = scmInfo['clone_url'];
      const name = getProjectName(origin);
      const revision = scmInfo.branch;
      const project: ProjectV2Source = { name, git: { remotes: { origin } } };
      if (revision) {
        project.git.checkoutFrom = { revision };
      }
      projects.push(project);
      devfile = Object.assign({ projects }, devfile);
    }
  }

  return devfile;
}

export function getProjectName(cloneUrl: string): string {
  let name = cloneUrl.split('/').reverse()[0].replace(/(.git)$/, '');
  name = name.replace(/([^-a-zA-Z0-9]+)/, '-');
  name = name.replace(/(^[-]+)/, '');
  name = name.replace(/([-]+$)/, '');
  if (name.length > PROJECT_NAME_MAX_LENGTH) {
    name = name.substr(0, PROJECT_NAME_MAX_LENGTH - 1);
  }

  return name;
}
