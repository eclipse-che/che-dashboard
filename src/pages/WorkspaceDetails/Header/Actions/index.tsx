/*
 * Copyright (c) 2018-2020 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';
import { CaretDownIcon } from '@patternfly/react-icons';
import React from 'react';
import WorkspaceDeleteAction, { WorkspaceDeleteAction as DeleteAction } from '../../../../components/Workspace/DeleteAction';
import { Actions } from '../../../../containers/WorkspaceDetails';
import { WorkspaceStatus } from '../../../../services/helpers/types';

import './Actions.styl';

type Props = {
  workspaceId: string;
  workspaceName: string;
  status: string | undefined;
  onAction: (action: Actions) => void;
};

type State = {
  isExpanded: boolean;
  isModalOpen: boolean;
}

export class HeaderActionSelect extends React.PureComponent<Props, State> {
  private readonly workspaceDeleteRef: React.RefObject<DeleteAction>;

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
      isModalOpen: false,
    };

    this.workspaceDeleteRef = React.createRef<DeleteAction>();
  }

  private handleToggle(isExpanded: boolean): void {
    if (this.state.isModalOpen) {
      return;
    }
    this.setState({ isExpanded });
  }

  private handleSelect(selected: Actions): void {
    this.setState({
      isExpanded: false,
    });
    this.props.onAction(selected);
  }

  private onDelete(): void {
    this.handleSelect(Actions.DELETE_WORKSPACE);
    this.setState({ isExpanded: true });
    this.workspaceDeleteRef.current?.onClick();
  }

  private onModalStatusChange(isModalOpen: boolean): void {
    this.setState({ isModalOpen });
    if (!isModalOpen && this.state.isExpanded) {
      this.setState({ isExpanded: false });
    }
  }

  private getDropdownItems(): React.ReactNode[] {
    const { workspaceId, status, workspaceName } = this.props;

    return [
      (<DropdownItem
        key={`action-${Actions.OPEN}`}
        onClick={() => this.handleSelect(Actions.OPEN)}>
        <div>{Actions.OPEN}</div>
      </DropdownItem>),
      (<DropdownItem
        key={`action-${Actions.OPEN_IN_VERBOSE_MODE}`}
        onClick={() => this.handleSelect(Actions.OPEN_IN_VERBOSE_MODE)}>
        <div>{Actions.OPEN_IN_VERBOSE_MODE}</div>
      </DropdownItem>),
      (<DropdownItem
        key={`action-${Actions.START_IN_BACKGROUND}`}
        isDisabled={status !== WorkspaceStatus[WorkspaceStatus.STOPPED]}
        onClick={() => this.handleSelect(Actions.START_IN_BACKGROUND)}>
        <div>{Actions.START_IN_BACKGROUND}</div>
      </DropdownItem>),
      (<DropdownItem
        key={`action-${Actions.STOP_WORKSPACE}`}
        isDisabled={status === WorkspaceStatus[WorkspaceStatus.STOPPED]}
        onClick={() => this.handleSelect(Actions.STOP_WORKSPACE)}>
        <div>{Actions.STOP_WORKSPACE}</div>
      </DropdownItem>),
      (<DropdownItem
        key={`action-${Actions.DELETE_WORKSPACE}`}
        isDisabled={status === WorkspaceStatus[WorkspaceStatus.STARTING] || status === WorkspaceStatus[WorkspaceStatus.STOPPING]}
        onClick={() => this.onDelete()}>
        <WorkspaceDeleteAction
          workspaceName={workspaceName}
          workspaceId={workspaceId}
          ref={this.workspaceDeleteRef}
          onModalStatusChange={isModalOpen => this.onModalStatusChange(isModalOpen)}
          status={status ? WorkspaceStatus[status] : WorkspaceStatus.STOPPED}>
          {Actions.DELETE_WORKSPACE}
        </WorkspaceDeleteAction>
      </DropdownItem>),
    ];
  }

  render(): React.ReactNode {
    const { workspaceId } = this.props;
    const { isExpanded } = this.state;

    return (
      <Dropdown
        className="workspace-action-selector"
        toggle={(
          <DropdownToggle
            data-testid={`${workspaceId}-action-dropdown`}
            onToggle={isExpanded => this.handleToggle(isExpanded)}
            toggleIndicator={CaretDownIcon}
            isPrimary>
            Actions
          </DropdownToggle>
        )}
        isOpen={isExpanded}
        position="right"
        dropdownItems={this.getDropdownItems()}
      />
    );
  }
}
