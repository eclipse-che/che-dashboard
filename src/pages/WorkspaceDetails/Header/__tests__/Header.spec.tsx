/*
 * Copyright (c) 2018-2020 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { WorkspaceStatus } from '../../../../services/helpers/types';
import { HeaderActionSelect } from '../Actions';
import Header from '../index';

jest.mock('../../../../containers/WorkspaceDetails', () => {
  return { Actions: {} };
});

describe('Workspace details header widget', () => {
  const workspaceName = 'test-workspace-name';
  const workspaceId = 'test-workspace-id';

  it('should render STOPPED status correctly', () => {
    const status = WorkspaceStatus.STOPPED;

    const component = createComponent(status, workspaceName, workspaceId);

    expect(getComponentSnapshot(component)).toMatchSnapshot();
  });

  it('should render RUNNING status correctly', () => {
    const status = WorkspaceStatus.RUNNING;

    const component = createComponent(status, workspaceName, workspaceId);

    expect(getComponentSnapshot(component)).toMatchSnapshot();
  });

});

function createComponent(
  workspaceStatus: WorkspaceStatus,
  workspaceName: string,
  workspaceId: string,
): React.ReactElement {
  return (
    <Header workspaceName={workspaceName} status={WorkspaceStatus[workspaceStatus]}>
      <HeaderActionSelect
        onAction={jest.fn()}
        workspaceId={workspaceId}
        workspaceName={workspaceName}
        status={WorkspaceStatus[workspaceStatus]} />
    </Header>
  );
}

function getComponentSnapshot(
  component: React.ReactElement
): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
  return renderer.create(component).toJSON();
}
