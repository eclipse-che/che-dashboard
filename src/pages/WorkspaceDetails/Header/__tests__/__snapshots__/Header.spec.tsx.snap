// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace details header widget should render RUNNING status correctly 1`] = `
<section
  className="pf-c-page__main-section pf-m-light"
>
  <nav
    aria-label="Breadcrumb"
    className="pf-c-breadcrumb"
    data-ouia-component-id={2}
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe={true}
  >
    <ol
      className="pf-c-breadcrumb__list"
    >
      <li
        className="pf-c-breadcrumb__item"
      >
        <a
          className="pf-c-breadcrumb__link"
          href="./#/workspaces"
          target={null}
        >
          Workspaces
        </a>
        <span
          className="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              transform=""
            />
          </svg>
        </span>
      </li>
      <li
        className="pf-c-breadcrumb__item"
      >
        test-workspace-name
      </li>
    </ol>
  </nav>
  <div
    className="pf-l-flex"
  >
    <div
      className=""
    >
      <h1
        className=""
        data-pf-content={true}
      >
        test-workspace-name
        <span
          className="pf-c-label pf-m-green"
        >
          <span
            className="pf-c-label__content"
          >
            <span
              className="pf-c-label__icon"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="green"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 64 1024 1024"
                width="1em"
              >
                <path
                  d="M512 109.714c-181.486 0-329.143 147.657-329.143 329.143s147.657 329.143 329.143 329.143c181.486 0 329.143-147.657 329.143-329.143s-147.657-329.143-329.143-329.143zM954.743 694.171c45.714-78.4 68.571-164.114 68.571-256.914s-22.857-178.514-68.571-256.914c-45.714-78.4-107.886-140.571-186.286-186.286s-164.114-68.571-256.914-68.571c-92.8 0-178.514 22.857-256.914 68.571s-140.571 107.886-186.286 186.286c-45.486 78.4-68.343 164.114-68.343 256.914s22.857 178.514 68.571 256.914c45.714 78.4 107.886 140.343 186.286 186.057s164.114 68.571 256.914 68.571c92.8 0 178.514-22.857 256.914-68.571 78.171-45.714 140.343-107.657 186.057-186.057zM677.257 829.714c-52.571 22.629-107.886 34.057-165.486 34.057-57.829 0-112.914-11.429-165.486-34.057s-97.829-53.029-136-90.971c-37.943-37.943-68.343-83.2-90.971-136-22.629-52.571-34.057-107.886-34.057-165.486 0-57.829 11.429-112.914 34.057-165.486s53.029-97.829 90.971-136c37.943-37.943 83.2-68.343 136-90.971 52.571-22.629 107.886-34.057 165.486-34.057 57.829 0 112.914 11.429 165.486 34.057s97.829 53.029 136 90.971c37.943 37.943 68.343 83.2 90.971 136 22.629 52.571 34.057 107.886 34.057 165.486 0 57.829-11.429 112.914-34.057 165.486s-53.029 97.829-90.971 136c-37.943 37.943-83.429 68.343-136 90.971z"
                  transform="rotate(180 0 512) scale(-1 1)"
                />
              </svg>
            </span>
            Running
          </span>
        </span>
      </h1>
    </div>
    <div
      className="pf-m-align-right"
    >
      <div
        className="pf-c-dropdown pf-m-align-right workspace-action-selector"
        data-ouia-component-id={3}
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe={true}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="pf-c-dropdown__toggle pf-m-primary"
          data-testid="test-workspace-id-action-dropdown"
          disabled={false}
          id="pf-toggle-id-1"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            className="pf-c-dropdown__toggle-text"
          >
            Actions
          </span>
          <span
            className="pf-c-dropdown__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 320 512"
              width="1em"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                transform=""
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`Workspace details header widget should render STOPPED status correctly 1`] = `
<section
  className="pf-c-page__main-section pf-m-light"
>
  <nav
    aria-label="Breadcrumb"
    className="pf-c-breadcrumb"
    data-ouia-component-id={0}
    data-ouia-component-type="PF4/Breadcrumb"
    data-ouia-safe={true}
  >
    <ol
      className="pf-c-breadcrumb__list"
    >
      <li
        className="pf-c-breadcrumb__item"
      >
        <a
          className="pf-c-breadcrumb__link"
          href="./#/workspaces"
          target={null}
        >
          Workspaces
        </a>
        <span
          className="pf-c-breadcrumb__item-divider"
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              transform=""
            />
          </svg>
        </span>
      </li>
      <li
        className="pf-c-breadcrumb__item"
      >
        test-workspace-name
      </li>
    </ol>
  </nav>
  <div
    className="pf-l-flex"
  >
    <div
      className=""
    >
      <h1
        className=""
        data-pf-content={true}
      >
        test-workspace-name
        <span
          className="pf-c-label"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          <span
            className="pf-c-label__content"
          >
            <span
              className="pf-c-label__icon"
            >
              <svg
                aria-hidden="true"
                fill="#151515"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8.001,0 C3.589,0 0,3.590 0,8 C0,12.410 3.590,16 8.001,16 C12.412,16 16,12.410
               16,8 C16,3.589 12.4125,0 8.001,0 Z M8,14 C4.690,14 2,11.310 2,8 C2,4.692 4.690,2
               8,2 C11.307,2 14,4.690 14,8 C14,11.309 11.307,14 8,14 Z"
                />
                <path
                  d="M8.001,3.5 C5.519,3.5 3.5,5.520 3.5,8 C3.5,10.481 5.520,12.5 8.000,12.5 C10.482,12.5 12.5,10.481
               12.5,8 C12.5,5.519 10.482,3.5 8.000,3.5 Z M8,10.75 C6.483,10.75 5.25,9.517 5.25,8 C5.25,6.484
               6.483,5.25 8,5.25 C9.516,5.25 10.75,6.483 10.75,8 C10.75,9.516 9.516,10.75 8,10.75 Z"
                />
              </svg>
            </span>
            Stopped
          </span>
        </span>
      </h1>
    </div>
    <div
      className="pf-m-align-right"
    >
      <div
        className="pf-c-dropdown pf-m-align-right workspace-action-selector"
        data-ouia-component-id={1}
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe={true}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="pf-c-dropdown__toggle pf-m-primary"
          data-testid="test-workspace-id-action-dropdown"
          disabled={false}
          id="pf-toggle-id-0"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            className="pf-c-dropdown__toggle-text"
          >
            Actions
          </span>
          <span
            className="pf-c-dropdown__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 320 512"
              width="1em"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                transform=""
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</section>
`;
